server/tests/models/user.test.js

```diff
@@ -1,45 +1,87 @@
 import chai, { expect } from 'chai';
 import User from '../../models/user';
+import Device from '../../models/device';
 
 chai.config.includeStack = true;
 
-describe('# User Model', () =>{
-  let user = new User({
-    name: 'Kaleb Davis',
-    email: 'abc@google.com',
-    role: 1
-  });
+describe('# User Model', () => {
+  describe('user creation', () => {
+    let user = new User({
+      name: 'Kaleb Davis',
+      email: 'abc@google.com'
+    });
 
-  before(() =>{
-    User.remove({}, () =>{});
-  });
-  
-  it('creates a user', (done) => {
-    User.create(user, (err, u) => {
-      expect(err).to.not.exist;
+    before(() =>{
+      User.remove({}, () => {});
+    });
 
-      expect(u.name).to.equal(user.name);
-      expect(u.email).to.equal(user.email);
-      expect(u.role).to.equal(user.role);
-      done();
+    it('creates a user', (done) => {
+      User.create(user, (err, u) => {
+        expect(err).to.not.exist;
+
+        expect(u.name).to.equal(user.name);
+        expect(u.email).to.equal(user.email);
+        expect(u.role).to.equal(user.role);
+        done();
+      });
+    });
+
+    it("throws an error when the email is not valid", (done) => {
+      const badEmail = 'hello';
+      user.email = badEmail;
+      User.create(user, (err, u) => {
+        expect(err).to.exist;
+        expect(err.errors['email'].message).to.equal(badEmail + ' is not a valid email address');
+        done();
+      });
     });
-  });
 
-  it("throws an error when the email is not valid", (done) => {
-    var badEmail = 'hello';
-    user.email = badEmail;
-    User.create(user, (err, u) => {
-      expect(err).to.exist;
-      expect(err.errors['email'].message).to.equal(badEmail + ' is not a valid email address');
-      done();
+    it("throws an error when a required field doesn't exist", (done) => {
+      user.name = null;
+      User.create(user, (err, u) => {
+        expect(err).to.exist;
+        done();
+      });
     });
   });
 
-  it("throws an error when a required field doesn't exist", (done) => {
-    user.name = null;
-    User.create(user, (err, u) => {
-      expect(err).to.exist;
-      done();
+  describe('user devices modifications', () => {
+    let user = new User({
+      name: 'Kaleb Davis',
+      email: 'abc@google.com'
+    });
+
+    const newDevice = new Device({
+      name: 'home phone',
+      type: 'phone',
+      contactInformation: '5555555555'
```

Comment:
If there is any validation, I don't think it should go in the user, it should be part of the Device model and it's validation